### DISCUSSION TITLE
I guess the readme needs a big update? (Latest master)

### DISCUSSION THREAD

**Discussion by KarimElsayad247** [H-M-H/Weylus#304] [General]
So I started working on an android client, I was working on the github released version but just today I switched to a locally built version, and already started facing some wrinkles:

**Building on Ubuntu 24.04 required installing some extra dependencies** 

At first I was trying to just `cargo build` which should build ffmpeg, but I had to install extra libs:

`livavformat-dev libavcodec-dev libswscale-dev libavdevice-dev libavfilter-dev libpostproc-dev`

But I still faced some linking problems so I gave up and used ` --features ffmpeg-system`

Then an extra linking problem because it didn't find `lxcb-dri3`, so ok had to install `libxcb-dri3-dev`

And finally it got built.

As for some other issues I faced:

- Now there is only 1 single port for both http and ws connections
- To connect to ws, I need to send a request to `/ws`

**Comment by H-M-H** [comment]
Hmm, Weylus should build fine on Ubuntu. Have a look at the Dockerfile for a working setup: https://github.com/H-M-H/Weylus/blob/master/docker/Dockerfile (it's Debian based but things should not be much different on Ubuntu). Regarding the Readme, yes the description about the ports used is outdated. But I will probably keep it that way until just before the next release so people downloading the binaries from the release page are not confused.

But feel free to contribute other improvements to the Readme.
